app/src/sharedTest/java/org/oppia/android/app/topic/practice/TopicPracticeFragmentTest.kt

```diff
@@ -90,322 +90,403 @@ class TopicPracticeFragmentTest {
 
   private var skillIdList = ArrayList<String>()
   private val internalProfileId = 0
-  private lateinit var activityScenario: ActivityScenario<TopicActivity>
 
-  @get:Rule
-  var activityTestRule: ActivityTestRule<TopicActivity> = ActivityTestRule(
-    TopicActivity::class.java,
-    /* initialTouchMode= */ true,
-    /* launchActivity= */ false
-  )
+  @Inject
+  lateinit var context: Context
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
 
   @Before
   fun setUp() {
-    FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext())
-    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
-    activityScenario = launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID)
     Intents.init()
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+    FirebaseApp.initializeApp(context)
     skillIdList.add("5RM9KPfQxobH")
     skillIdList.add("B39yK4cbHZYI")
   }
 
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+    Intents.release()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
   @Test
-  // TODO(#973): Fix TopicPracticeFragmentTest
-  @Ignore
   fun testTopicPracticeFragment_loadFragment_displaySubtopics_startButtonIsInactive() {
     launchTopicActivityIntent(internalProfileId, FRACTIONS_TOPIC_ID).use {
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withText(TopicTab.getTabForPosition(2).name),
           isDescendantOfA(withId(R.id.topic_tabs_container))
         )
       ).perform(click())
-      onView(withId(R.id.master_skills_text_view))
-        .check(
-          matches(
-            withText(
-              R.string.topic_practice_master_these_skills
-            )
+      testCoroutineDispatchers.runCurrent()
+      onView(
```

Comment:
For here & tests below: can we move these clicks/matches into specific helper functions to simplify the test bodies? Look at StateFragmentTest for inspiration on how helpers can provide semantic context to make tests smaller, much more readable, and make helper function code highly reusable.

Reply:
done